### ISSUE TITLE
Integration test decorators are firing during the import phase of running an unrelated test suite

### ISSUE THREAD

**Issue by bgeltz** [geopm/geopm#1107]
In Python2, importing modules using the ```from <PACKAGE> import <MODULE>``` syntax is causing the ```__init__.py``` to fire.  Once init.py starts, it tries to issue ```from test_integration.geopm_test_integration import *``` which starts executing random test decorators.  An example:

```
$ python -m pdb ./test_frequency_hint_usage.py
> /home/bgeltz/git/geopm/test_integration/test_frequency_hint_usage.py(40)<module>()
-> """
(Pdb) break /home/bgeltz/git/geopm/scripts/geopmpy/launcher.py:535
Breakpoint 1 at /home/bgeltz/git/geopm/scripts/geopmpy/launcher.py:535
(Pdb) c
> /home/bgeltz/git/geopm/scripts/geopmpy/launcher.py(535)run()
-> pid = subprocess.Popen(argv_mod, env=self.environ(),
(Pdb) argv_mod
'srun -N 1 -n 1 -J geopm_allocation_test --cpu-freq=Performance -- stat /sys/devices/system/cpu/cpu0/cpufreq/cpuinfo_min_freq'
(Pdb) bt
  /usr/lib64/python2.7/bdb.py(400)run()
-> exec cmd in globals, locals
  <string>(1)<module>()
  /home/bgeltz/git/geopm/test_integration/test_frequency_hint_usage.py(48)<module>()
-> from test_integration import geopm_context
  /home/bgeltz/git/geopm/test_integration/__init__.py(40)<module>()
-> from test_integration.geopm_test_integration import *
  /home/bgeltz/git/geopm/test_integration/geopm_test_integration.py(100)<module>()
-> class TestIntegration(unittest.TestCase):
  /home/bgeltz/git/geopm/test_integration/geopm_test_integration.py(1196)TestIntegration()
-> @util.skip_unless_cpufreq()
  /home/bgeltz/git/geopm/test_integration/util.py(111)skip_unless_cpufreq()
-> geopm_test_launcher.allocation_node_test(test_exec, dev_null, dev_null)
  /home/bgeltz/git/geopm/test_integration/geopm_test_launcher.py(100)allocation_node_test()
-> launcher.run(stdout, stderr)
> /home/bgeltz/git/geopm/scripts/geopmpy/launcher.py(535)run()
-> pid = subprocess.Popen(argv_mod, env=self.environ(),
(Pdb)
```

(I set a breakpoint set in launcher.py just before the Popen() call.)

The problem above starts with ```from test_integration import geopm_context```.  If this were changed to just ```import geopm_context``` the problem goes away because ```__init__.py``` does not fire.  I think we can simply change all of the imports in the integration test dir to *not* use ``` from <MODULE> ...``` and this will go away.

The above example illustrates that we start in ```hint_usage.py```.  We execute until we get to the first ```from <MODULE>``` import, then ```__init__.py``` fires.  Once that happens the main ```integration_test.py``` is imported and one of the test's decorators starting on line 1196 fires.

This code path is erroneous and has nothing to do with where we started (i.e. the imports for the ```hint_usage.py```).

**Comment by bgeltz** [comment]
While testing my changes for #547, I put a debug print in the run() method of the launcher.py.  I'm printing out argv_mod after "BRG =>".

It looks like all the test decorators fire when util.py is imported:
```
$ GEOPM_KEEP_FILES=yes ./geopm_test_integration.py -fv TestIntegration.tes
t_count
BRG => srun -N 1 -n 1 -J geopm_allocation_test --cpu-freq=Performance -- stat /sys/devices/system/cpu/cpu0/cpufreq/c
puinfo_min_freq
BRG => srun -N 1 -n 1 -J geopm_allocation_test --cpu-freq=Performance -- mkdir -vp /etc/geopm
BRG => srun -N 1 -n 1 -J geopm_allocation_test --cpu-freq=Performance -- mkdir -vp /etc/geopm
BRG => srun -N 1 -n 1 -J geopm_allocation_test --cpu-freq=Performance -- ldconfig --print-cache
BRG => srun -N 1 -n 1 -J geopm_allocation_test --cpu-freq=Performance -- stress-ng -h
BRG => srun -N 1 -n 1 -J geopm_allocation_test --cpu-freq=Performance -- stat /sys/devices/system/cpu/cpu0/cpufreq/c
puinfo_min_freq
BRG => srun -N 1 -n 1 -J geopm_allocation_test --cpu-freq=Performance -- stat /sys/devices/system/cpu/cpu0/cpufreq/c
puinfo_min_freq
BRG => srun -N 1 -n 1 -J geopm_allocation_test --cpu-freq=Performance -- stat /sys/devices/system/cpu/cpu0/cpufreq/c
puinfo_min_freq
BRG => srun -N 1 -n 1 -J geopm_allocation_test --cpu-freq=Performance -- stat /sys/devices/system/cpu/cpu0/cpufreq/c
puinfo_min_freq
BRG => srun -N 1 -n 1 -J geopm_allocation_test --cpu-freq=Performance -- mkdir -vp /etc/geopm
BRG => srun -N 1 -n 1 -J geopm_allocation_test --cpu-freq=Performance -- mkdir -vp /etc/geopm
BRG => srun -N 1 -n 1 -J geopm_allocation_test --cpu-freq=Performance -- ldconfig --print-cache
BRG => srun -N 1 -n 1 -J geopm_allocation_test --cpu-freq=Performance -- stress-ng -h
test_count (__main__.TestIntegration) ... BRG => srun -N 1 -n 1 -J geopm_allocation_test --cpu-freq=Performance -- $
eopmread MSR::PKG_POWER_LIMIT:PL1_POWER_LIMIT board 0
BRG => srun -N 1 -n 1 -J geopm_allocation_test --cpu-freq=Performance -- geopmread MSR::PERF_CTL:FREQ board 0
BRG => srun -N 1 -n 1 --cpu-freq=Performance lscpu
srun: Loaded msrsave restore plugin.
srun: Loaded geopm plugin.

srun -N 1 -n 1 --cpu-freq=Performance -- geopmread --cache

BRG => srun -N 1 -n 1 --cpu-freq=Performance -- geopmread --cache
srun: Loaded msrsave restore plugin.
srun: Loaded geopm plugin.
BRG => srun -N 1 -n 1 --cpu-freq=Performance -- lscpu --hex
srun: Loaded msrsave restore plugin.
srun: Loaded geopm plugin.
BRG => srun -N 1 -n 5 --cpu-bind v,mask_cpu:0x2,0xffc,0x3ff000,0x3ff000000,0xffc00000000 --export=LD_PRELOAD=libgeop
m.so,ALL -I30 -t 10 -J test_count --cpu-freq=Performance -- geopmbench --verbose test_count_app.config
... output continues normally from here...
```

**Comment by bgeltz** [comment]
This was resolved by #1215.

**Comment by bgeltz** [comment]
While #1215 did fix my issue with ```--skip-launch```, doing a real run is still showing this problem.

In the run I just tried, I had set ```OMP_NUM_THREADS=DNC``` for something else I'm trying.  This shows that the problem persists:

```
$ OMP_NUM_THREADS=DNC ${WORKSPACE}/geopm/integration/test/test_power_balancer.py --show-details
Traceback (most recent call last):
  File "/usr/workspace/wsb/intelpwr/intelCRADA/geopm/integration/test/test_power_balancer.py", line 46, in <module>
    from integration.test import geopm_context
  File "/usr/workspace/wsb/intelpwr/intelCRADA/geopm/integration/test/__init__.py", line 40, in <module>
    from integration.test.geopm_test_integration import *
  File "/usr/workspace/wsb/intelpwr/intelCRADA/geopm/integration/test/geopm_test_integration.py", line 83, in <module>
    class TestIntegration(unittest.TestCase):
  File "/usr/workspace/wsb/intelpwr/intelCRADA/geopm/integration/test/geopm_test_integration.py", line 1052, in TestIntegration
    @util.skip_unless_cpufreq()
  File "/usr/workspace/wsb/intelpwr/intelCRADA/geopm/integration/test/util.py", line 190, in skip_unless_cpufreq
    geopm_test_launcher.allocation_node_test(test_exec, dev_null, dev_null)
  File "/usr/workspace/wsb/intelpwr/intelCRADA/geopm/integration/test/geopm_test_launcher.py", line 99, in allocation_node_test
    job_name="geopm_allocation_test", quiet=True)
  File "/usr/WS2/intelpwr/intelCRADA/geopm/scripts/geopmpy/launcher.py", line 87, in create
    host_file, reservation=reservation, quiet=quiet)
  File "/usr/WS2/intelpwr/intelCRADA/geopm/scripts/geopmpy/launcher.py", line 898, in __init__
    reservation=reservation, quiet=quiet)
  File "/usr/WS2/intelpwr/intelCRADA/geopm/scripts/geopmpy/launcher.py", line 429, in __init__
    self.cpu_per_rank = int(os.environ.get('OMP_NUM_THREADS', '1'))
ValueError: invalid literal for int() with base 10: 'DNC'
```

The back trace shows that we jump from test_power_balancer.py to __init__.py via trying to import geopm_context, then we jump into geopm_test_integration.py then decorators start firing.
